rotkehlchen/chain/ethereum/modules/aave/v2/decoder.py

```diff
@@ -0,0 +1,267 @@
+            notes += f' on behalf of {on_behalf_of}'
+        for event in decoded_events:
+            if (
+                event.location_label == user and
+                event.balance.amount == amount and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                if event.event_type == HistoryEventType.SPEND:
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                    event.notes = notes
+                    event.counterparty = CPT_AAVE_V2
+                else:
+                    event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                    resolved_asset = event.asset.resolve_to_asset_with_symbol()
+                    event.notes = f'Receive {event.balance.amount} {resolved_asset.symbol} from AAVE v2'  # noqa: E501
+                    event.counterparty = CPT_AAVE_V2
+                # Set counterparty for both events
+                event.counterparty = CPT_AAVE_V2
+
+    def _decode_withdrawal(
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+    ) -> None:
+        """Decode aave v2 withdrawal event"""
+        user = hex_or_bytes_to_address(tx_log.topics[2])
+        if self.base_tools.is_tracked(user) is False:
+            return
+        to = hex_or_bytes_to_address(tx_log.topics[3])
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(tx_log.data),
+            asset=token,
+        )
+        notes = f'Withdraw {amount} {token.symbol} from AAVE v2'
+        if to != user:
+            notes += f' to {to}'
+        for event in decoded_events:
+            if (
+                event.balance.amount == amount and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                if (
+                    event.location_label == to and
+                    event.event_type == HistoryEventType.RECEIVE
+                ):
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                    event.location_label = user
+                    event.notes = notes
+                else:
+                    event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                    resolved_asset = event.asset.resolve_to_asset_with_symbol()
+                    event.notes = f'Return {amount} {resolved_asset.symbol} to AAVE v2'  # noqa: E501
+                # Set counterparty for both events
+                event.counterparty = CPT_AAVE_V2
+
+    def _decode_borrow(
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+    ) -> None:
+        """Decode aave v2 borrow event"""
+        on_behalf_of = hex_or_bytes_to_address(tx_log.topics[2])
```

Comment:
perhaps as mentioned above the `on_behalf_of` should also be checked for tracking and if tracked continue in the function?

Reply:
Yes, makes sense